### PR TITLE
jobs: Supply a reason in all cases

### PR DESCRIPTION
**Pull request by kalikiana**
Also allow API calls to specify a reason.

See: https://progress.opensuse.org/issues/111590#note-5

### CODE REVIEW

File: lib/OpenQA/WebAPI/Controller/API/V1/Job.pm [hunk 1]
```diff
@@ -971,13 +971,14 @@ Used for both apiv1_cancel and apiv1_cancel_jobs
 =cut
 
 sub cancel {
-    my ($self) = @_;
+    my ($self, $reason = undef) = @_;
```
Comment (Martchus) [inline lib/OpenQA/WebAPI/Controller/API/V1/Job.pm:974]: I suppose you need to use signatures for the default-assignment to work.

**Review by Martchus — COMMENTED** [review]
Generally a good idea.

**Review by perlpunk — APPROVED** [review]

File: lib/OpenQA/Schema/ResultSet/DeveloperSessions.pm [hunk 2]
```diff
@@ -66,7 +66,7 @@ sub unregister {
         sub {
             my $session = $self->find({job_id => $job_id}) or return 0;
             my $job = $session->job or return 0;
-            return $job->cancel(OpenQA::Jobs::Constants::USER_CANCELLED);
+            return $job->cancel(OpenQA::Jobs::Constants::USER_CANCELLED, "closing developer session");
```
Comment (okurz) [inline lib/OpenQA/Schema/ResultSet/DeveloperSessions.pm:69]:
```suggestion
            return $job->cancel(OpenQA::Jobs::Constants::USER_CANCELLED, 'closing developer session');
```

**Review by okurz — APPROVED** [review]

**Comment by perlpunk** [comment]
the OBS API was in maintenance mode this morning at some point, that could explain why new builds haven't been triggered: https://build.opensuse.org/package/show/devel:openQA:GitHub:os-autoinst:openQA:PR-4681/openQA
maybe do a push again?

**Comment by kalikiana** [comment]
> the OBS API was in maintenance mode this morning at some point, that could explain why new builds haven't been triggered: https://build.opensuse.org/package/show/devel:openQA:GitHub:os-autoinst:openQA:PR-4681/openQA maybe do a push again?

Ack.

Also in the meantime I realized why coverage was missing. So I added a unit test for `jobs/cancel`.

**Pull request merged**